github/repos_pages.go

```diff
@@ -8,15 +8,23 @@ package github
 import (
 	"context"
 	"fmt"
+	"strings"
 )
 
 // Pages represents a GitHub Pages site configuration.
 type Pages struct {
-	URL       *string `json:"url,omitempty"`
-	Status    *string `json:"status,omitempty"`
-	CNAME     *string `json:"cname,omitempty"`
-	Custom404 *bool   `json:"custom_404,omitempty"`
-	HTMLURL   *string `json:"html_url,omitempty"`
+	URL       *string      `json:"url,omitempty"`
+	Status    *string      `json:"status,omitempty"`
+	CNAME     *string      `json:"cname,omitempty"`
+	Custom404 *bool        `json:"custom_404,omitempty"`
+	HTMLURL   *string      `json:"html_url,omitempty"`
+	Source    *PageOptions `json:"source,omitempty"`
+}
+
+// PageOptions represents options to enable/disable pages
```

Comment:
Looking at this again, I'm thinking that maybe this should be called `PageSiteSource` instead of `PageOptions`.

What do you think, @gauntface ?

Reply:
I would have probably gone with `PagesSource` but agree that `PageOptions` doesn't feel right here.

I'd go for `PagesSource` instead of `PageSiteSource` to make it match up with `Pages` struct. If it was `PagesSite` and `PagesSiteSource` then I'd be 👍